test/span.js

```diff
@@ -182,6 +186,127 @@ describe('span should', () => {
+                span.addTags({
+                    'tagKeyTwo': 'tagValueTwo'
+                });
+                span.log({'logkeyOne': 'logValueOne'});
+
+                tracer._sampler = new ConstSampler(o.sampling);
+                span.setOperationName('sampled-span');
+                span.finish();
+
+                assert.isOk(deepEqual(span._tags[0], {key: 'tagKeyOne', value: 'tagValueOne'}));
+                assert.isOk(deepEqual(span._tags[1], {key: 'tagKeyTwo', value: 'tagValueTwo'}));
+                assert.isOk(deepEqual(span._logs[0].fields[0], {key: 'logkeyOne', value: 'logValueOne'}));
+                assert.equal(reporter.spans.length, o.reportedSpans);
+            });
+        });
+
+        describe('span sampling finalizer', () => {
+            it ('should trigger on a span that inherits a sampling decision', () => {
+                let childSpan = tracer.startSpan('child-span', {childOf: span});
+                assert.isOk(span.context().samplingFinalized);
+                assert.isOk(childSpan.context().samplingFinalized);
+            });
+
+            it ('should trigger on a span that set the sampling priority', () => {
+                span.setTag(opentracing.Tags.SAMPLING_PRIORITY, 1);
+                assert.isOk(span.context().samplingFinalized);
+
+                let unsampledSpan = tracer.startSpan('usampled-span');
+                unsampledSpan.setTag(opentracing.Tags.SAMPLING_PRIORITY, -1);
+                assert.isOk(unsampledSpan.context().samplingFinalized);
+            });
+
+            it ('should trigger on a finish()-ed span', () => {
+                span.finish();
+                assert.isOk(span.context().samplingFinalized);
+            });
+
+            it ('should trigger after calling setOperationName', () => {
+                span.setOperationName('fry');
+                assert.isOk(span.context().samplingFinalized);
+            });
+
+            it ('should trigger when a span is injected into headers', () => {
+                let headers = {};
+                tracer.inject(span.context(), opentracing.FORMAT_HTTP_HEADERS, headers);
+
+                assert.isOk(span.context().samplingFinalized);
+            });
+        });
+
+        it ('isWriteable returns true if not finalized, or the span is sampled', () => {
+            tracer = new Tracer(
+                'test-service-name',
+                new InMemoryReporter(),
+                new ConstSampler(false),
+                { logger: new MockLogger() }
+            );
+            let unFinalizedSpan = tracer.startSpan('unFinalizedSpan');
+            assert.isOk(unFinalizedSpan._isWriteable());
+
+            tracer._sampler = new ConstSampler(true);
+            let sampledSpan = tracer.startSpan('sampled-span');
+            assert.isOk(sampledSpan._isWriteable());
+        });
+
+        it ('2nd setOperationName should add sampler tags to span, and change operationName', () => {
```

Comment:
I meant 2nd because the I want to make sure on the 2nd set operationName that the tags change, and the operation name changes.